### PR TITLE
Feat/cheatcode

### PR DESCRIPTION
**Pull request by SheepFromHeaven**
The goal of this change is to give the technical onborders an example answer for the questions, so they know what is to be expected of the answer.

### CODE REVIEW

File: src/fundamentals-2.md [hunk 1]
```diff
@@ -81,12 +102,19 @@ Build a so called 'call to action', consisting of a image, text and button. The
...
+<questions-component>
+  <question-component>What is BEM and what are its benefits?</question-component>
+  <answer-component>BEM stands for Block, Element, Modifier. This methology seperates styles in Blocks, which are standalone entities, Elements, which are just a part of a block and modifiers, which are used to flag Blocks or Elements. This clear structure helps to build clearly understandable CSS/Sass code.</answer-component>
+  <question-component>What are the benefits of sass compared to plain CSS?</question-component>
+  <answer-component>
+  In sass you, can declare variables, mixins, functions and baseclasses which you then can use all over your styles. The helps to avoid repetition and keeps your codebase clean (DRY principle)</answer-component>
+  <question-component>Unmodified sass code can be used in web browsers. True of false?</question-component>
+  <answer-component>False! Sass always needs to be compiled to css to be run in the browser.</answer-component>
+  <question-component>What is a mixin in sass?</question-component>
+  <answer-component>A mixin is a reusable block of sass that inserts rules into the place where you use it.</answer-component>
```
Comment (dweiger) [inline src/fundamentals-2.md]: Maybe adding an example for a mixin and a function?

File: src/advanced-components.md [hunk 2]
```diff
@@ -30,17 +30,37 @@ Rewrite your Accordion or, if you want a challenge, a useful component using an
...
+  <question-component>What is immutable vs. mutable?</question-component>
+  <answer-component>An immutable value will never be changed. It is constant. Rather then changing the value we create a new values that represents the new state of the old one. A mutable value on the other hand can be change at will, which you will never know the current value of it.</answer-component>
+  <question-component>When should you use a mutable data structure?</question-component>
+  <answer-component>Never! Well, only if there I defenitely no way to solve your problem in an immutable way</answer-component>
+  <question-component>Can you write your own filter functions using reduce?</question-component>
+  <answer-component>Yes!</answer-component>
+  <question-component>How do you use a spread operator and what is it good for?</question-component>
+  <answer-component>A spread operator will return all the bare values of the spread itrable object. It is written as three dots: <code>...</code></answer-component>
```
[the remark this answers is not in the document]
Comment (smfonseca) [reply src/advanced-components.md]:
```suggestion
<answer-component>The spread operator allows an expression to be expanded in places where multiple elements/variables/arguments are expected. It is written as three dots: <code>...</code>. A simple example: 
<code>
const num1 = { a: '1', b: '2', c: '3'};
const num2 = { ...num1, d: '4', e: '5'};
// num2 = { a: '1', b: '2', c: '3', d: '4', e: '5' }
</code>
</answer-component>
```

File: src/advanced-components.md [hunk 3]
```diff
@@ -30,17 +30,37 @@ Rewrite your Accordion or, if you want a challenge, a useful component using an
...
-7. Do we have alternatives to REST?
-8. What is REST? What describes it best?
-9. Do we have alternatives to HTTP1.1
-10. How does a HTTP request work?
-11. How does a file upload work?
+<questions-component>
+  <question-component>Why are we using TS?</question-component>
+  <answer-component>Typescript enhances Javascript and gives it typesafety. This helps us to readicate a lot of potential bugs already on build time as well as keeping a clear pictures of the data strctures being used in the project.</answer-component>
+  <question-component>What is immutable vs. mutable?</question-component>
+  <answer-component>An immutable value will never be changed. It is constant. Rather then changing the value we create a new values that represents the new state of the old one. A mutable value on the other hand can be change at will, which you will never know the current value of it.</answer-component>
```
[the remark this answers is not in the document]
Comment (SheepFromHeaven) [reply src/advanced-components.md:37]:
```suggestion
  <answer-component>An immutable value will never be changed. It is constant. Rather then changing the value we create a new value that represents the new state of the old one. A mutable value on the other hand can be change at will, which you will never know the current value of it.</answer-component>
```

File: src/fundamentals-2.md (hunk 1 above)
Comment (SheepFromHeaven) [reply src/fundamentals-2.md:114]: I think an onboarder should be able to show some code like this somewhere. Therefore it doesn't make sense to me.

**Pull request merged**